packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -199,6 +215,24 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     this.assertComponentElement(this.firstChild, { tagName: 'div', attrs: { class: 'ember-view' }, content: 'hello' });
   }
 
+  ['@test class property on components can be dynamic']() {
+    this.registerComponent('foo-bar', { template: 'hello' });
+
+    this.render('{{foo-bar class=(if fooBar "foo-bar")}}', {
+      fooBar: true
+    });
+
+    this.assertComponentElement(this.firstChild, { content: 'hello', attrs: { 'class': classes('ember-view foo-bar') } });
+
+    this.runTask(() => this.rerender());
+
+    this.assertComponentElement(this.firstChild, { content: 'hello', attrs: { 'class': classes('foo-bar ember-view') } });
```

Comment:
The order doesn't matter: http://stackoverflow.com/questions/1321692/how-to-specifiy-the-order-of-css-classes

Also, yep, the `classes` helper sorts them anyway; which is why I suggested changing it to be consistent to not mislead other people in thinking that there is any significance between the two lines